R/step_adjust_latency.R

```diff
@@ -0,0 +1,274 @@
+#'   sophisticated methods of extrapolation.
+#'   - `"extend_lags"`: per `epi_key` and `predictor`, adjusts the lag so that
+#'   the shortest lag at predict time is at the last observation. E.g. if the
+#'   lags are `c(0,7,14)` for data that is 3 days latent, the actual lags used
+#'   become `c(3,10,17)`.
+#' @param fixed_latency either a positive integer, or a labeled positive integer
+#'   vector. Cannot be set at the same time as `fixed_asof`.   If non-`NULL`,
+#'   the amount to offset the ahead or lag by. If a single integer, this is used
+#'   for all columns; if a labeled vector, the labels must correspond to the
+#'   base column names.  If `NULL`, the latency is the distance between the
+#'   `epi_df`'s `max_time_value` and either the `fixed_asof` or the `epi_df`'s
+#'   `as_of` field.
+#' @param fixed_asof either a date of the same kind used in the `epi_df`, or
+#'   NULL. Cannot be set at the same time as `fixed_latency`. If a date, it
+#'   gives the date from which the forecast is actually occurring. If `NULL`,
+#'   the `as_of` is determined either from `fixed_latency` or automatically.
+#' @param role For model terms created by this step, what analysis role should
+#'   they be assigned? `lag` is default a predictor while `ahead` is an outcome.
+#'   It should be correctly inferred and not need setting
+#' @param trained A logical to indicate if the quantities for preprocessing have
+#'   been estimated.
+#' @param columns A character string of column names to be adjusted; these
+#'   should be the original columns, and not the derived ones
+#' @param default Determines what fills empty rows
+#'   left by leading/lagging (defaults to NA).
+#' @param skip A logical. Should the step be skipped when the
+#'  recipe is baked by [bake()]? While all operations are baked
+#'  when [prep()] is run, some operations may not be able to be
+#'  conducted on new data (e.g. processing the outcome variable(s)).
+#'  Care should be taken when using `skip = TRUE` as it may affect
+#'  the computations for subsequent operations.
+#' @param id A unique identifier for the step
+#' @template step-return
+#'
+#' @details The step assumes that the pipeline has already applied either
+#'   `step_epi_ahead` or `step_epi_lag` depending on the value of
+#'   `"method"`, and that `step_epi_naomit` has NOT been run.
+#'
+#' The `prefix` and `id` arguments are unchangeable to ensure that the code runs
+#' properly and to avoid inconsistency with naming. For `step_epi_ahead`, they
+#' are always set to `"ahead_"` and `"epi_ahead"` respectively, while for
+#' `step_epi_lag`, they are set to `"lag_"` and `"epi_lag`, respectively.
+#'
+#' @family row operation steps
+#' @rdname step_adjust_latency
+#' @export
+#' @examples
+#' r <- epi_recipe(case_death_rate_subset) %>%
+#'   step_epi_ahead(death_rate, ahead = 7) %>%
+#'   #   step_adjust_latency(method = "extend_ahead") %>%
+#'   step_epi_lag(death_rate, lag = c(0, 7, 14))
+#' r
```

Comment:
It would help me to flesh this out a bit. Maybe something with latency and see the results with and without?

Reply:
For now, do the ones in the tests generally address that? I still need to update the layer functions to use dates correctly, but was waiting until these steps are relatively stable. Once I have those, I'll definitely expand the `@examples`, and then write a vignette